pkg/odo/genericclioptions/runnable.go

```diff
@@ -3,21 +3,39 @@ package genericclioptions
 import (
 	"flag"
 	"fmt"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"k8s.io/klog"
 	"os"
+	"time"
 
 	"github.com/openshift/odo/pkg/log"
 	"github.com/openshift/odo/pkg/odo/util"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
+var (
+	segmentClient *segment.Client
+	//ctx context.Context
+)
+
 type Runnable interface {
 	Complete(name string, cmd *cobra.Command, args []string) error
 	Validate() error
 	Run() error
 }
 
 func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
+	//ctx = telemetry.NewContext(context.Background())
+	startTime := time.Now()
+	cfg, _ := preference.New()
+	var err error
+	// Initiate the segment client
+	if segmentClient, err = segment.NewClient(cfg); err != nil {
+		klog.Fatal(err.Error())
```

Comment:
As telemetry is an auxiliary system - we never fail on it. Even if there is an error - propogate it to upper layers or just log it as info. We never fail if segment is not available.

Reply:
that is a good point. failing to send or create telemetry should not effect odo functionality.
The error should be just logged `klog.V(4).Infof`